lib/src/markdown_processor.dart

```diff
@@ -560,6 +467,122 @@ ModelElement _findRefElementInLibrary(String codeRef, Warnable element,
+      if (!_ConsiderIfConstructor(codeRef, modelElement)) continue;
+      results.add(packageGraph.findCanonicalModelElementFor(
+          modelElement.element,
+          preferredClass: preferredClass));
+    }
+  }
+  results.remove(null);
+}
+
+void _findWithinRefElementCache(Set<ModelElement> results, String codeRefChomped, PackageGraph packageGraph, String codeRef) {
+  // We now need the ref element cache to keep from repeatedly searching [Package.allModelElements].
+  // But if not, look for a fully qualified match.  (That only makes sense
+  // if the codeRef might be qualified, and contains periods.)
+  if (results.isEmpty &&
+      codeRefChomped.contains('.') &&
+      packageGraph.findRefElementCache.containsKey(codeRefChomped)) {
+    for (final ModelElement modelElement
+        in packageGraph.findRefElementCache[codeRefChomped]) {
+      if (!_ConsiderIfConstructor(codeRef, modelElement)) continue;
+      // For fully qualified matches, the original preferredClass passed
+      // might make no sense.  Instead, use the enclosing class from the
+      // element in [_findRefElementCache], because that element's enclosing
+      // class will be preferred from [codeRefChomped]'s perspective.
+      results.add(packageGraph.findCanonicalModelElementFor(
+          modelElement.element,
+          preferredClass: modelElement.enclosingElement is Class
+              ? modelElement.enclosingElement
+              : null));
+    }
+  }
+  results.remove(null);
+}
+
+void _findWithinTryClasses(Set<ModelElement> results, Class preferredClass, Warnable element, String codeRefChomped, String codeRef, PackageGraph packageGraph) {
+  if (results.isEmpty) {
+    // Maybe this is local to a class.
+    // TODO(jcollins-g): tryClasses is a strict subset of the superclass chain.  Optimize.
+    List<Class> tryClasses = [preferredClass];
+    Class realClass = tryClasses.first;
+    if (element is Inheritable) {
+      Inheritable overriddenElement = element.overriddenElement;
+      while (overriddenElement != null) {
+        tryClasses.add(
+            (element.overriddenElement as EnclosedElement).enclosingElement);
+        overriddenElement = overriddenElement.overriddenElement;
+      }
+    }
+
+    for (Class tryClass in tryClasses) {
+      if (tryClass != null) {
+        _getResultsForClass(
+            tryClass, codeRefChomped, results, codeRef, packageGraph);
+      }
+      results.remove(null);
+      if (results.isNotEmpty) break;
+    }
+
+    if (results.isEmpty && realClass != null) {
+      for (Class superClass
+          in realClass.publicSuperChain.map((et) => et.element as Class)) {
+        if (!tryClasses.contains(superClass)) {
+          _getResultsForClass(
+              superClass, codeRefChomped, results, codeRef, packageGraph);
+        }
+        results.remove(null);
+        if (results.isNotEmpty) break;
+      }
+    }
+  }
+  results.remove(null);
```

Comment:
... And why again here and in the super chain loop above?

Reply:
Necessary inside the loop because we're again using "set is not empty" as a proxy for no results, and sometimes null gets added instead of a result.